### ISSUE TITLE
Create alternative for accuracy as reader study metric

### ISSUE THREAD

**Issue by Kicksta** [comic/grand-challenge.org#1878]
**Is your feature request related to a problem? Please describe.**
I have created a reader study where readers have to provide probabilities between 0 and 100 for the presence of an abnormality.  The final score is based on whether the exact number in the ground truth (either 0 or 100) matches exactly with the score by the reader. This is not favorable as this not reflects the actual performance of the reader.

**Describe the solution you'd like**
I see two options:
1) Be able to pass a range as ground truth. (e.g. if score between 0 and 49 --> correct)
2) Be able to output other metrics than accuracy, such as AUCROC as in this case it would better reflect the performance.
Please, choose what you think is best.

**Comment by jmsmkn** [comment]
My guess is that option 2 fits in better with what we have. The scoring function is provided here: https://github.com/comic/grand-challenge.org/blob/16c70c9f6c41c03cb51b8504379a5a64aad88f62/app/grandchallenge/reader_studies/models.py#L1181-L1186

We would need:
- add `roc_auc_score` to these options https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html
- add a choice field on `Question` to switch between the scoring method with the default of accuracy
- use the scoring method when required
- add the scoring method to the question edit form, make this non-editable after create (or, if it changes re-score everything that relies on it via a celery task, quite some work for that)

**Comment by Kicksta** [comment]
I found a way around it for now, but will at some point need it (and maybe also include some other metrics). So also here, could reopen if I'll start on it?

**Comment by jmsmkn** [comment]
That sounds fine.

**Comment by Kicksta** [comment]
Created a branch for this and did step 1 to 4. Will discuss the work + step 5 with @MikeOverkamp-diag tomorrow.

**Comment by Kicksta** [comment]
@MikeOverkamp-diag, I continued working on it but could use your help here… I just pushed the latest version.
-	In `conftest.py`, `_reader_study_with_gt` I defined questions and answers which represent (if I understand correctly) the ground truth values. They could not all be set to True, because `sklearn.roc_auc_score` needs two classes (otherwise aucroc is always 1). So I used a formula here (`(i + j) % 2`) resulting in `[0,1,1,0,0,1]`.
-	In `test_models.py` I define again answers which are float (`1 / (i + j + 1)`) resulting in `[1, 1/2, 1/2, 1/3, 1/3, 1/4]`. 
-	These two should result in a roc_auc_score of 0.556. Currently it outputs `score__sum` and `score__avg` of 0.0.
-	I wonder if `score__sum `and `score__avg `make sense as you can not calculate an auc for an individual case. And that makes me wonder if it will work with the whole setup as is, as it is based on case level scores, right?
Could you have a look and see how we should continue?
